public/app/core/logsModel.ts

```diff
@@ -418,7 +418,13 @@ export function logSeriesToLogsModel(logSeries: DataFrame[], queries: DataQuery[
 
       const hasUnescapedContent = !!message.match(/\\n|\\t|\\r/);
 
-      const searchWords = series.meta && series.meta.searchWords ? series.meta.searchWords : [];
+      // Data sources that set up searchWords on backend use meta.custom.searchWords
+      // Data sources that set up searchWords trough frontend can use meta.searchWords
+      const searchWords = series?.meta?.custom?.searchWords
```

Comment:
nitpick: could this maybe be simplified into:
```ts
series.meta?.custom?.searchWords ?? series.meta?.searchWords ?? [];
```

Reply:
Fixed in https://github.com/grafana/grafana/pull/63924/commits/80ef4c9455fac1f9c3033f81229dd61fa915eefa